src/types/global.d.ts

```diff
@@ -1,15 +1,19 @@
- interface Developer{
+
+interface Developer<T = {}>{
     developerName: string;
     age: number;
     isEmployed: boolean;
     skills: string[];
     projects: Project[];
     experience: number;
     email: string;
+    [key: string]: T;
```

Comment:
give reason for this addition

Reply:
When we are adding a new property to the developer object, it was giving error because it is of Developer type which doesn't allow addition of any new properties. Therefore, I did this. I am not too sure of this approach. If there is any alternative or improvement, please let me know.

Reply:
let's discuss this in the demo